src/core/ddsi/include/dds/ddsi/q_time.h

```diff
@@ -53,9 +53,9 @@ int valid_ddsi_timestamp (nn_ddsi_time_t t);
 DDS_EXPORT nn_wctime_t now (void);       /* wall clock time */
 DDS_EXPORT nn_mtime_t now_mt (void);     /* monotonic time */
 DDS_EXPORT nn_etime_t now_et (void);     /* elapsed time */
-DDS_EXPORT void mtime_to_sec_usec (int * __restrict sec, int * __restrict usec, nn_mtime_t t);
-DDS_EXPORT void wctime_to_sec_usec (int * __restrict sec, int * __restrict usec, nn_wctime_t t);
-DDS_EXPORT void etime_to_sec_usec (int * __restrict sec, int * __restrict usec, nn_etime_t t);
+DDS_EXPORT void mtime_to_sec_usec (int32_t * __restrict sec, int32_t * __restrict usec, nn_mtime_t t);
```

Comment:
Switching to an unsigned type would deal gracefully with wrap around after 2106 or thereabouts (even if it is only used for printing). Secondly, the format strings need to updated as well — from %d to whichever type is used.

Reply:
For now I'll focus on getting FreeRTOS support done first. Switching to an unsigned int later on is the same amount of work, so I feel it's not a priority right now. I am wondering which format string you mean wrt these functions. I though I fixed all of them. Maybe I'm overlooking something?